### PR TITLE
Fix wrong memory addresses due to not accounted GROUPR offset

### PR DESCRIPTION
**Pull request by kernie**
GROUPR writes gendf data in a compact way and let us know, what the offset is.

The dlayxs part of CCCR loops over the full energy group range in container variable `a` where it should loop only until the offset. Thereby, on chance uninitialized or invalid memory may be addressed and may lead at best to a crash or simply runs without error but resulting in values like 1.E+44 for delayed spectra.

This commit introduces the offset and corrects 2 loops. Moreover,  a check is introduced if the requested memory fits into the `a` variable. The `a` size variable `isiza` was moved to a parameter statement and renamed to account for 4 byte and 8 byte equivalence'd variables.

### CODE REVIEW

**Comment by kernie** [comment]
To see the erroneous bevavior, one can take [input.cccr_u232_361.txt](https://github.com/njoy/NJOY2016/files/9573570/input.cccr_u232_361.txt) with ENDF6 data of U-232. It uses the SHEM-361 group structure.

**Review by whaeck — CHANGES_REQUESTED** [review]
Just a few minor questions that we should address before approving this.

I have asked @nathangibson14 to have a look if he has some time - but I think these changes are pretty straight forward.

File: ReleaseNotes.md [hunk 1]
```diff
@@ -26,7 +27,7 @@ The following minor changes and fixes are included in this version:
 A large number of compiler warnings have now been resolved (unused variables, large static arrays converted into allocatable arrays, etc.). There are still a lot of warnings but their number is heavily reduced. For source files that were corrected in this way, the remaining warnings relate to equality comparisons for real values, unused dummy arguments in subroutines and potential 0 indices into arrays (in all cases, if statements prevented this from happening).
 
 ## [NJOY2016.66](https://github.com/njoy/NJOY2016/pull/226)
-This constitutes a major update of NJOY to accommodate the latest IAEA photonuclear data library (which uses non-isotropic photon distributions) and the mixed mode elastic scattering ENDF format (adopted in the ENDF format in November 2020).
+This constitutes a major update of NJOY to aommodate the latest IAEA photonuclear data library (which uses non-isotropic photon distributions) and the mixed mode elastic scattering ENDF format (adopted in the ENDF format in November 2020).
```
[the remark this answers is not in the document]
Comment (kernie) [reply ReleaseNotes.md:30]: Yes, looks like ghost-typing...

File: src/ccccr.f90 [hunk 2]
```diff
@@ -3125,7 +3125,12 @@ subroutine dldata(nin)
          call contio(nin,0,0,e(1),nb,nw)
       enddo
       if (mfh.eq.5) then
+         ! number of delayed neutron groups
          ndg=nint(e(3))
+         call contio(nin,0,0,e(1),nb,nw)
+         ! offset of energy groups where no delayed data is given
+         ! position 9 of gendf tape mf5 mt455
+         ig1=nint(e(3))-1
```
[the remark this answers is not in the document]
Comment (kernie) [reply src/ccccr.f90:3133]: I think the 1 is save. I'm not aware of a higher energy limit for fission in general (but this is out of my theoretical knowledge).

If I understand GROUPR correctly the first position in the delayed fission matrix holds the decay constant of the precursor family (GROUPR.f90:720). That is read before. And then we need to subtract 1 to get the real offset. I don't know whether it is the ideal place to do so, but at some point this is necessary.

File: tests/77/input [hunk 3]
```diff
@@ -0,0 +1,137 @@
+ 'moder'
+ 20  -21
+ 'reconr'
+ -21 -22
+ 'pendf tape for  U 232 from jeff331'/
```
[the remark this answers is not in the document]
Comment (kernie) [reply tests/77/input:5]: true

File: tests/77/input [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+ 3 452 'nu-t' /
+ 3 455 'nu-d' /
+ 3 456 'nu-p' /
+ 5 18 'chi ' /
+ 5 455 'chid' /
+ 6 /
+ 0/
+ 0/
+ 'ccccr'
+ 25 00  00  30
```
[the remark this answers is not in the document]
Comment (kernie) [reply tests/77/input:132]: I'll check that

[the remark this answers is not in the document]
Comment (kernie) [reply tests/77/input]: I removed test 77, since you are right. Test 76 covers it.

**Review by whaeck — APPROVED** [review]
I think this is good to go. I'm waiting for some feedback from @nathangibson14 before I merge but I think we won't need more changes here.

I did verify the specs for the delayed neutron data in the GENDF file and that -1 does indeed need to be taken into account. Putting the comment there makes it easier to understand as well.

**Pull request merged**